### ISSUE TITLE
 must contain a value for "import_path"

### ISSUE THREAD

**Issue by maylad31** [gunthercox/ChatterBot#969]
My code for logic adapter in django gives me the above mentioned error. Please help.
code:

       'logic_adapters' : [
          {
             "import_path": 'chatterbot.logic.BestMatch',

          }, 

        {

            "import_path": 'chatterbot.logic.LowConfidenceAdapter',
            "threshold": 0.90,
            "default_response": 'OhNo'
        }
        ],

**Comment by rodneyhayes** [comment]
Try:

logic_adapters=[
      {
         "import_path": 'chatterbot.logic.BestMatch',

      }, 
      {

        "import_path": 'chatterbot.logic.LowConfidenceAdapter',
        "threshold": 0.90,
        "default_response": 'OhNo'
      }
],

**Comment by maylad31** [comment]
hi thanks one more issue i am facing in django is that i am getting the previous sentence in the list as a response when i was running in flask it was alright. Please help me.

**Comment by rodneyhayes** [comment]
What are you using to get responses?

**Comment by maylad31** [comment]
this is my code that i have in settings.py file


CHATTERBOT = {
    'name': 'Django ChatterBot Example',
    'trainer': 'chatterbot.trainers.ListTrainer',
    'training_data': [
        #here i have my data
],
 'django_app_name': 'django_chatterbot'
}

**Comment by vkosuri** [comment]
which version of chatterbot are you using? To know version ``python -m chatterbot --version``

**Comment by maylad31** [comment]
0.7.6
I am not getting any problem when i am using flask.

**Comment by taiwotman** [comment]
Did you have this problem when you `python manage.py train `?
If yes,  comment out(#) the whole **logic_adapters** in your settings.py and uncomment after your training is done.

**Comment by taiwotman** [comment]
@maylad31, can you confirm asap if your issue has been solved. You are free to re-open.
@vkosuri can you this close #969?

**Comment by maylad31** [comment]
okay i will try as you say. yes please close this issue as i have shifted my app on flask. However i will still try with django and if the problem persists i will reopen. Thanks.

**Comment by Super-Fly** [comment]
I have the same issue with Django -> `python manage.py train`
```
CHATTERBOT = {
    'name': 'My Bot',
    'logic_adapters': [
        'chatterbot.logic.MathematicalEvaluation',
        {
            "import_path": "chatterbot.logic.BestMatch",
            "statement_comparison_function": "chatterbot.comparisons.levenshtein_distance",
            "response_selection_method": "chatterbot.response_selection.get_first_response"
        },
        {
            'import_path': 'chatterbot.logic.LowConfidenceAdapter',
            'threshold': 0.65,
            'default_response': 'I am sorry, but I do not understand.'
        },
        {
            'import_path': 'chatterbot.logic.SpecificResponseAdapter',
            'input_text': 'owl',
            'output_text': 'you have correct owl emoji, come visit this URL'
        }
    ],
    'preprocessors': [
        'chatterbot.preprocessors.clean_whitespace'
    ],
    'django_app_name': 'My Bot'
}
```

The error is in chatterbot/utils.py file on line 54, trying to get import_path but the structure is not like this

**Comment by vkosuri** [comment]
Yes, It is reproducing I'll tag it as Bug

**Comment by phunsukwangdu** [comment]
yes ,i am too facing same problems :(

**Comment by taiwotman** [comment]
@phunsukwangdu, It is a bug as @gunthercox pointed.  Hope the fix will be out soon. At the moment, the workaround for me as provided in my earlier comment was to do your training while commenting out logic adapters with/out the threshold and after training uncomment it back.

**Comment by phunsukwangdu** [comment]
@taiwotman thx man!!! it worked!!

**Comment by gunthercox** [comment]
This issue has been fixed as a part of #969

**Comment by Gingerguo** [comment]
<img width="536" alt="screenshot 2018-02-25 17 27 14" src="https://user-images.githubusercontent.com/14199542/36647294-6c3b7132-1a51-11e8-805e-ecb62616e731.png">
<img width="297" alt="screenshot 2018-02-25 17 27 22" src="https://user-images.githubusercontent.com/14199542/36647295-6c483f0c-1a51-11e8-90e4-4bd4d7c51c94.png">
